cmd/abapEnvironmentCloneGitRepo.go

```diff
@@ -0,0 +1,178 @@
+	}
+	jsonBody := []byte(`{"sc_name":"` + repositoryName + `"}`)
+	resp, err = abaputils.GetHTTPResponse("POST", cloneConnectionDetails, jsonBody, client)
+	if err != nil {
+		err = abaputils.HandleHTTPError(resp, err, "Could not clone the Repository / Software Component "+repositoryName+" with branch "+branchName, uriConnectionDetails)
+		return uriConnectionDetails, err
+	}
+	defer resp.Body.Close()
+	log.Entry().WithField("StatusCode", resp.Status).WithField("repositoryName", repositoryName).WithField("branchName", branchName).Info("Triggered Clone of Repository / Software Component")
+
+	// Parse Response
+	var body abaputils.CloneEntity
+	var abapResp map[string]*json.RawMessage
+	bodyText, errRead := ioutil.ReadAll(resp.Body)
+	if errRead != nil {
+		return uriConnectionDetails, err
+	}
+	json.Unmarshal(bodyText, &abapResp)
+	json.Unmarshal(*abapResp["d"], &body)
+	if reflect.DeepEqual(abaputils.CloneEntity{}, body) {
+		log.Entry().WithField("StatusCode", resp.Status).WithField("repositoryName", repositoryName).WithField("branchName", branchName).Error("Could not Clone the Repository / Software Component")
+		err := errors.New("Request to ABAP System not successful")
+		return uriConnectionDetails, err
+	}
+
+	expandLog := "?$expand=to_Execution_log,to_Transport_log"
+
+	// The entity "Clones" does not allow for polling. To poll the progress, the related entity "Pull" has to be called
+	// While "Clones" has the key fields UUID, SC_NAME and BRANCH_NAME, "Pull" only has the key field UUID
+	tempURI := strings.Replace(body.Metadata.URI, "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Clones", "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull", 1)
+	pollingURI := strings.Replace(tempURI, ",sc_name='"+repositoryName+"',branch_name='"+branchName+"'", "", 1)
+
+	uriConnectionDetails.URL = pollingURI + expandLog
+	return uriConnectionDetails, nil
+}
+
+func getRepositories(config *abapEnvironmentCloneGitRepoOptions) []abaputils.Repository {
+	var repositories = make([]abaputils.Repository, 0)
+	if config.Repositories != "" {
+		descriptor, err := abaputils.ReadAddonDescriptor(config.Repositories)
+		if err == nil {
+			repositories = descriptor.Repositories
+		}
+	}
+	if config.RepositoryName != "" && config.BranchName != "" {
+		repositories = append(repositories, abaputils.Repository{Name: config.RepositoryName, Branch: config.BranchName})
+	}
+	return repositories
+}
+
+func convertConfig(config *abapEnvironmentCloneGitRepoOptions) abaputils.AbapEnvironmentOptions {
+	subOptions := abaputils.AbapEnvironmentOptions{}
+
+	subOptions.CfAPIEndpoint = config.CfAPIEndpoint
+	subOptions.CfServiceInstance = config.CfServiceInstance
+	subOptions.CfServiceKeyName = config.CfServiceKeyName
+	subOptions.CfOrg = config.CfOrg
+	subOptions.CfSpace = config.CfSpace
+	subOptions.Host = config.Host
+	subOptions.Password = config.Password
+	subOptions.Username = config.Username
+	return subOptions
+}
```

Comment:
Wouldn't it be helpful to implement this func in the abapUtils or manageGitRepo package as it is used in the other steps as well?

Reply:
I'm not sure if this could be changed to a reusable function. For each step the input type looks different.